guildwars2/account.py

```diff
@@ -297,6 +297,124 @@ def refined_armor_scan(a, b=ids_refined_envoy_armor):
+                    {"name": "Mursaat Overseer", "type": "single", "id": 3321},
+                    {"name": "Mursaat Overseer CM", "type": "single", "id": 3287},
+                    {"name": "Samarog", "type": "single", "id": 3347},
+                    {"name": "Samarog CM", "type": "single", "id": 3342},
+                    {"name": "Deimos", "type": "single", "id": 3364},
+                    {"name": "Deimos CM", "type": "single", "id": 3292}
+                ],
+                "Hall of Chains": [
+                    {"name": "Soulless Horror", "type": "single", "id": 4004},
+                    {"name": "Soulless Horror CM", "type": "single", "id": 3993},
+                    {"name": "River of Souls", "type": "single", "id": 4010},
+                    {"name": "Statues of Grenth", "type": "all", "ids": [3998, 4036, 4038]},
+                    {"name": "Voice in the Void", "type": "single", "id": 4016},
+                    {"name": "Voice in the Void CM", "type": "single", "id": 3979}
+                ],
+                "Fractals": [
+                    {"name": "Nightmare CM", "type": "single", "id": 3180},
+                    {"name": "Shattered Observatory CM", "type": "single", "id": 3465}
+                ],
+        }
+
+        # Create a list of lists of ids.
+        achievement_ids = [[x["id"]] if x["type"] == "single" else x["ids"]
+                for x in chain.from_iterable(wings.values())]
+        # Flatten it.
+        achievement_ids = [str(x) for x in chain.from_iterable(achievement_ids)]
+
+        try:
+            doc = await self.fetch_key(user, scopes)
+            endpoint = "account/achievements?ids=" + ",".join(achievement_ids)
+            results = await self.call_api(endpoint, user, ["progression"])
+        except APINotFound as e:
+            # Not Found is returned by the API when none of the searched
+            # achievements have been completed yet.
+            results = []
+        except APIError as e:
+            return await self.error_handler(ctx, e)
+
+        def is_killed(boss):
+            # If the achievement is finished, the encounter was completed
+            if boss["type"] == "single":
+                for achievement in results:
+                    if achievement["id"] == boss["id"] and achievement["done"]:
+                        return "+";
+                # The achievement is not in the list or isn't done
+                return "-"
+            # If all achievements were completed, the encounter was completed
+            if boss["type"] == "all":
+                for id in boss["ids"]:
+                    found = False
+                    for achievement in results:
+                        if achievement["id"] == id and achievement["done"]:
+                            found = True
+                            break
+                    if not found:
+                        return "-"
+                return "+"
```

Comment:
just early return if found, no need for the `found` variable and the last if statement

Reply:
That cannot be done in this case as we are checking whether all achievements have been completed. I have simplified it with any().